### PR TITLE
[mergebot] post comment on finishing

### PR DESCRIPTION
**Pull request by zengk95**
This makes the bot comment to ping the user whenever the merge or revert finishes completing. Also does a bit of refactoring. 

Addreses https://github.com/pytorch/test-infra/issues/392

### CODE REVIEW

**Review by janeyx99 — COMMENTED** [review]
What are the benefits of the refactoring? It doesn't see more clear/less repetition but adds another layer of function calls.

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -909,6 +913,7 @@ def post_comment(msg: str) -> None:
     msg += f" due to {reason}\n" if reason is not None else "\n"
     repo.amend_commit_message(msg)
     repo.push(pr.default_branch(), dry_run)
+    post_comment(f"@{pr.get_author()} your PR has been successfully reverted.")
```
[the remark this answers is not in the document]
Comment (zengk95) [reply .github/scripts/trymerge.py:916]: you don't need it in here cause the dry_run part gets passed in on line 880.

File: .github/scripts/trymerge.py [hunk 2]
```diff
@@ -729,6 +729,7 @@ def merge_into(self, repo: GitRepo, *, force: bool = False, dry_run: bool = Fals
             self.merge_ghstack_into(repo, force, comment_id=comment_id)
 
         repo.push(self.default_branch(), dry_run)
+        gh_post_comment(self.org, self.project, self.pr_num, f"@{self.get_author()} your PR has been successfully merged.", dry_run)
```
Comment (seemethere) [inline .github/scripts/trymerge.py:732]: Can we just edit the last comment we posted? I feel like adding more comments like this is semi-annoying

**Comment by suo** [comment]
I prefer new comments because it’s easier to track timeline and they
generate notifs

File: .github/scripts/trymerge.py (hunk 2 above)
Comment (seemethere) [reply .github/scripts/trymerge.py:732]: Agree with @suo's assessment over tracking timeline / generating notifications

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**